### DISCUSSION TITLE
Additional SIB support. SIB10/11/12

### DISCUSSION THREAD

**Discussion by c-aman** [srsran/srsRAN_4G#1255] [Ideas]
Hi, Can anyone tell about which files to be changed to add support for new SIB messages sib10, sib11, sib12. I want enB to transmit them when it receives write-rep req from MME over s1ap.

Thanks in advance!

**Comment by SAIRISAN123** [comment]
@c-aman have you done this setup. I am also need to do this I am able send message from MME. But enB S1AP issue.

**Reply by c-aman** [reply]
https://github.com/open5gs/open5gs/discussions/2743#discussion-5875141
right.

**Reply by c-aman** [reply]
I looked for adding these SIB support but I was not able to connect handset with the srsRAN but I think its already done by some guys but that code is not merged into main branch.
You can check https://github.com/learning-lte/srsLTE_cmas_etws

**Reply by SAIRISAN123** [reply]
@c-aman I have already implemented this part too. But I think the above implementation is not related to s1AP. that like manually injecting messages from eNodeB. But that's working fine. Correct me If I am wrong

If you have any idea share with me about s1AP

**Reply by c-aman** [reply]
I haven't looked the eNB code but if its manually injecting message then it won' work.
One way is, if you have eNB logs you can check where eNB is receiving this S1AP request and in that file you need to add a parser (use already built-in, if any) and extract the parameters and pass it to sib configuration to use these values instead if hard-coded.

**Reply by c-aman** [reply]
or first to check if anyone already have done it. If I came across any such repo I'll post it here.

**Reply by SAIRISAN123** [reply]
@c-aman Thank you. I'll update you.

**Reply by Andrea99-beep** [reply]
Have you solved your problems? Cause I'm stuck too
